### DISCUSSION TITLE
Hydrate before Migrate?

### DISCUSSION THREAD

**Discussion by ouya99** [pmndrs/zustand#1469] [Q&A]
Hello!

i would like to migrate data that had been persisted in localStorage, but once i do

`migrate: (persistedState,version)  => { 'code to migrate from one version to another' }
`
as shown in docs, my persistedState data is always empty. For testing purposes I fill localStorage with data that should be migrated and launch the web app. So i would like this data to be in persistedState to be migrated to next version.

How would you solve this issue? I read something about hydration. Could i tell zustand to hydrate the persistedState with localStorage data before migration starts?

Thank you for help!

**Comment by dai-shi** [comment]
Can anyone help? (I'm not sure if I understand the problem.)
